android/java/org/chromium/chrome/browser/vpn/InAppPurchaseWrapper.java

```diff
@@ -0,0 +1,177 @@
+        this.context = context;
+        billingClient = BillingClient.newBuilder(context)
+                                .enablePendingPurchases()
+                                .setListener(purchasesUpdatedListener)
+                                .build();
+
+        connectToBillingService();
+    }
+
+    public void connectToBillingService() {
+        if (!billingClient.isReady()) {
+            billingClient.startConnection(billingClientStateListener);
+        }
+    }
+
+    public BillingClient getBillingClient() {
+        return billingClient;
+    }
+
+    public void querySkuDetailsAsync(List<String> skuList) {
+        SkuDetailsParams params = SkuDetailsParams.newBuilder()
+                                          .setSkusList(skuList)
+                                          .setType(BillingClient.SkuType.SUBS)
+                                          .build();
+        billingClient.querySkuDetailsAsync(params, (billingResult, skuDetailsList) -> {
+            if (billingResult.getResponseCode() == OK && skuDetailsList != null) {
+                for (SkuDetails skuDetails : skuDetailsList) {
+                    skusWithSkuDetails.put(skuDetails.getSku(), skuDetails);
+                }
+            }
+        });
+    }
+
+    public List<Purchase> queryPurchases() {
+        return billingClient.queryPurchases(SUBS).getPurchasesList();
+    }
+
+    public void purchase(Activity activity, SkuDetails skuDetails) {
+        BillingFlowParams billingFlowParams =
+                BillingFlowParams.newBuilder().setSkuDetails(skuDetails).build();
+
+        int responseCode =
+                billingClient.launchBillingFlow(activity, billingFlowParams).getResponseCode();
+    }
+
+    public void processPurchases(List<Purchase> purchases) {
+        for (Purchase purchase : purchases) {
+            if (purchase.getPurchaseState() == Purchase.PurchaseState.PURCHASED) {
+                acknowledgePurchase(purchase);
+            }
+        }
+    }
+
+    private void acknowledgePurchase(Purchase purchase) {
+        AcknowledgePurchaseParams acknowledgePurchaseParams =
+                AcknowledgePurchaseParams.newBuilder()
+                        .setPurchaseToken(purchase.getPurchaseToken())
+                        .build();
+        if (!purchase.isAcknowledged()) {
+            billingClient.acknowledgePurchase(acknowledgePurchaseParams, billingResult -> {
+                if (billingResult.getResponseCode() == OK) {
+                    BraveVpnPrefUtils.setBraveVpnBooleanPref(
+                            BraveVpnPrefUtils.PREF_BRAVE_VPN_SUBSCRIPTION_PURCHASE, true);
+                    BraveVpnUtils.openBraveVpnProfileActivity(context);
+                    Toast.makeText(context, "Subscription successfully consumed", Toast.LENGTH_LONG)
+                            .show();
+                } else {
+                    Toast.makeText(context, "Failed to acknowledge purchase :" + billingResult,
```

Comment:
localization

Reply:
just pushed with a commit which would address it